opennms-config/src/main/java/org/opennms/netmgt/config/SnmpPeerFactory.java

```diff
@@ -109,9 +109,7 @@ public class SnmpPeerFactory implements SnmpAgentConfigFactory {
      */
     private static AtomicBoolean s_loaded = new AtomicBoolean(false);
 
-    private final ReadWriteUpdateLock m_globalLock = new ReentrantReadWriteUpdateLock();
-    private final Lock m_readLock = m_globalLock.updateLock();
```

Comment:
For some reason we used an update lock here... but you can only hold 1 concurrently, so it defeats the purpose of the read lock.